nucypher/acumen/perception.py

```diff
@@ -42,22 +42,27 @@ class FleetSensor:
     log = Logger("Learning")
     FleetState = namedtuple("FleetState", ("nickname", "icon", "nodes", "updated", "checksum"))
 
-    def __init__(self):
+    def __init__(self, domain: str):
+        self.domain = domain
         self.additional_nodes_to_track = []
         self.updated = maya.now()
         self._nodes = OrderedDict()
         self._marked = defaultdict(list)  # Beginning of bucketing.
         self.states = OrderedDict()
 
-    def __setitem__(self, key, value):
-        self._nodes[key] = value
+    def __setitem__(self, checksum_address, node_or_sprout):
+        if node_or_sprout.domain == self.domain:
+            self._nodes[checksum_address] = node_or_sprout
 
-        if self._tracking:
-            self.log.info("Updating fleet state after saving node {}".format(value))
-            self.record_fleet_state()
+            if self._tracking:
+                self.log.info("Updating fleet state after saving node {}".format(node_or_sprout))
```

Comment:
Should we move this to `debug` instead of `info`?

Reply:
Oh, uhh... I guess so?  Is this one really noisy?  I don't think we really roll with `_tracking` on, and I presume that if someone / something does (like a test) that it's for a reason, and worth logging as info.  But... I dunno?  You like debug better for this?

Reply:
I was uncertain about its noisiness, but was thinking about it in the same spirit given the network has some ~1800 nodes. I didn't know the default value for `_tracking` (incorrectly assumed True), but if by default it is `False` then info log level is probably fine.